### ISSUE TITLE
Can not set random mob textures for modded 1.12.2

### ISSUE THREAD

**Issue by InfinitysDaughter** [sp614x/optifine#2728]
(I know. Its a confusing title.)

Optifine version: 1.12.2_HD_U_E3
Forge: 14.23.5.2838

I am trying to make new textures via a resource pack for the kitty mob in the mo'creatures mod. I want to set a few options for skins with different weights.

The textures I set to replace the default show up just fine, however no matter what I try, the alternative textures do not load. I even tested by setting the default to weight 0 and the alternative at 1 but the default is still the only one that loads.

**Comment by cap10zeus** [comment]
Yup, getting the same issue. Spent 4 hours trying to figure out what was going on, should have looked here to see it has been a long-time issue.
